src/components/nodes/pool/pool.vue

```diff
@@ -558,18 +559,19 @@ export default {
             invalidPool = null;
           }
 
-          this.paper.drawBackground({ color: invalidNodeColor });
-          element.component.allowSetNodePosition = false;
+          store.commit('preventSavingState');
+          this.paperManager.setInvalidGirdColor();
```

Comment:
Gird (https://www.merriam-webster.com/dictionary/gird) or Grid?

Reply:
Also, I wonder if this could be better named. Callers of this method shouldn't care about what's happening, all they should care about is that the paper manager is managing the invalid state for them. how about `setStateInvalid()` and `setStateValid()` instead?